### PR TITLE
Fixed: The code inside the #if SP_DEBUG > 2 did not compile.

### PR DESCRIPTION
**Pull request by kmokstad**
B is therefore changed into a single vector (of dimension nnod*ncomp)
instead of an array of vectors of length nnod. The equation solver
can handle this format, no need to loop over the RHS outside here.
Also some indentation fixes here and there.

### CODE REVIEW

File: src/ASM/LR/ASMu3Drecovery.C [hunk 1]
```diff
@@ -558,88 +553,80 @@ bool ASMu3D::faceL2projection (const DirichletFace& edge,
...
+        for (il = 0; il < edge.MNPC[ie].size(); il++) // local i-index
+          if ((ig = 1+edge.MNPC[ie][il]) > 0)         // global i-index
+          {
+            for (jl = 0; jl < edge.MNPC[ie].size(); jl++) // local j-index
+              if ((jg = 1+edge.MNPC[ie][jl]) > 0)         // global j-index
+                A(ig,jg) += fe.N[il]*fe.N[jl]*fe.detJxW;
+
+            RealArray val = values.getValue(X);
+            for (size_t k = 0; k < m; k++)
+              B(ig+(k-1)*n) += fe.N[il]*val[k]*fe.detJxW;
```
Comment (akva2) [inline src/ASM/LR/ASMu3Drecovery.C]: k*n, not (k-1)*n

**Comment by akva2** [comment]
except the above, this seems fine. please pick https://github.com/akva2/IFEM/tree/edge_proj (and review) while you are at it.

File: src/ASM/LR/ASMu3Drecovery.C (hunk 1 above)
Comment (kmokstad) [inline src/ASM/LR/ASMu3Drecovery.C]: Fixed.

**Comment by kmokstad** [comment]
Didn't pick that branch yet, since not sure how many commits there are relevant. But I reviewed it briefly and added a comment. So if this one does not break any tests it is ready to be merged (I need it, just for compiling Debug mode).

**Comment by akva2** [comment]
Then merge

**Pull request closed without merging**